packages/@angular/cli/commands/new.ts

```diff
@@ -39,7 +39,8 @@ const NewCommand = Command.extend({
       type: Boolean,
       default: false,
       aliases: ['lc'],
-      description: 'Automatically link the `@angular/cli` package.'
+      description: 'Automatically link the `@angular/cli` package.',
+      hidden: true
```

Comment:
It's only used for development purposes. Sometimes people think it's an install optimization but it really isn't.